kernel/constr.ml

```diff
@@ -650,6 +650,17 @@ let always_true _ _ = true
 let rec eq_constr_nounivs m n =
   (m == n) || compare_head_gen (fun _ -> always_true) always_true eq_constr_nounivs m n
 
+(** Universes *)
+
+let universe_levels_of_term c =
+  let rec aux acc c =
+    match kind c with
+    | Const (_, u) | Ind (_, u) | Construct (_, u) ->
+       Univ.LSet.union acc (Univ.Instance.levels u)
+    | Sort (Sorts.Type u) -> Univ.LSet.union acc (Univ.Universe.levels u)
+    | k -> fold aux acc k
+  in aux Univ.LSet.empty c
```

Comment:
Isn't this function the same as Univops.universes_of_constr? Except this uses LSet.union where the other uses a fold over the instance.

Reply:
Right, that's duplicated